pkg/registryclient/client.go

```diff
@@ -165,14 +168,36 @@ func WithTracing() Option {
 	}
 }
 
-// BuildRemoteOption builds remote.Option based on client.
-func (c *client) BuildRemoteOption(ctx context.Context) remote.Option {
-	return remote.WithRemoteOptions(
+// BuildCosignRemoteOption builds remote.Option for cosign client.
+func (c *client) BuildCosignRemoteOption(ctx context.Context) (remote.Option, error) {
+	gcrRemoteOpts, err := c.BuildGCRRemoteOption(ctx)
```

Comment:
Why builds gcrRemoteOpts in cosign?

Reply:
Kindly ping @vishal-chdhry.

Reply:
I have refactored it a bit [`81745c3` (#8494)](https://github.com/kyverno/kyverno/pull/8494/commits/81745c31fe04addacb2f97c5756cb5c17dc03ec4)